packages/widget/src/components/amount-selector/amount-selector.ts

```diff
@@ -1,22 +1,32 @@
 import type { Resource } from '@buildwithsygma/sygma-sdk-core';
 import { utils, type BigNumber } from 'ethers';
-import type { HTMLTemplateResult } from 'lit';
+import type { HTMLTemplateResult, PropertyDeclaration } from 'lit';
 import { html } from 'lit';
 import { customElement, property, state } from 'lit/decorators.js';
 import { classMap } from 'lit/directives/class-map.js';
 import { when } from 'lit/directives/when.js';
+import type { ApiPromise } from '@polkadot/api';
 import { networkIconsMap } from '../../assets';
-import { TokenBalanceController } from '../../controllers/wallet-manager/token-balance';
-import { tokenBalanceToNumber } from '../../utils/token';
-import type { DropdownOption } from '../common/dropdown/dropdown';
 import { DEFAULT_ETH_DECIMALS } from '../../constants';
+import {
+  BALANCE_UPDATE_KEY,
+  TokenBalanceController
+} from '../../controllers/wallet-manager/token-balance';
+
+import type { DropdownOption } from '../common/dropdown/dropdown';
 import { BaseComponent } from '../common/base-component';
+import { tokenBalanceToNumber } from '../../utils/token';
 import { styles } from './styles';
 
 @customElement('sygma-resource-selector')
 export class AmountSelector extends BaseComponent {
   static styles = styles;
 
+  @property({
+    type: Object
+  })
+  sourceSubstrateProvider?: ApiPromise;
```

Comment:
I would remove this. Not really scalable if we start adding new chains. I would replace with sourceDomainId

Reply:
@mpetrunic since amount-selector does not have access to `fungible-transfer-controller` state, having this as domain id will require passing down domain id -> parachain id mapping to use correct substrate provider, it is simpler to pass source parachain id IMO

Reply:
I'm trying to keep it generic for adding new chains. I don't think amount selector needs this it just passes it into TokenBalance controller which has access to substrate provider context

Reply:
I understand but token balance controller does not have the context of domain <> parachain mapping. 

Since we are storing providers w.r.t parachain ID, providing a domain ID would require fetching parachain ID using domain ID and it wouldn't be possible as SDK is initialiazed in Fungible transfer controller. We can use `sourceDomainConfig` which may or may not include parachain id and is generic enough